*This document goes over the steps to annotate and check the metagenome bins for completeness and contamination.*

Load required packages.

```{r}
library(tidyverse)
library(GEOquery)
library(Biostrings)
library(dplyr)
```

Import data. Note that this data is not publicly available yet, so I'm getting it from my Google Drive.

```{r}
# set path to all data
path <- "~/Library/CloudStorage/GoogleDrive-jwinter2@uw.edu/Shared drives/Rocap Lab/Project_ODZ_Marinimicrobia_Jordan/"
```

Find bins that are Marinimicrobia. These bins are MAGs (metagenome assembled genomes), and I'll find which are Marinimicrobia by using the CATBAT consensus results of what each contig is. Since I know what microbe each contig is, I'll look for bins where the majority of contigs are Marinimicrobia.

```{r}
# open file
classification <- read.csv(paste0(path, "CAT-BAT/1058_P1_2018_585_0.2um/1058_P1_2018_585_0.2um_assembly_plus.contig2classification_names.txt"), sep = "\t")

# get only those labeled as Marinimicrobia
classification <- classification[str_detect(classification$species, "Marinimicrobia"),]

# add this classification file to the repo
write.csv(classification, "../output/CATBAT_marinimicrobia.csv")
```

Now that I have a classification file and my bins, I can find bins where the majority of reads and contigs are Marinimicrobia

```{r}
# get Marinimicrobia bins
classification <- read.csv("../data/CATBAT_all.txt", sep = "\t")

# run through each binning method
bin_path <- list.dirs("../data/Bins")
mar_bins <- NULL
a <- 0

for (i in 2:7){
  
  pathname <- paste0(bin_path[i], "/")
  bins <- list.files(pathname)
  
  for (binname in bins){
    print(binname)
    bin_pathname <- paste0(pathname, binname)
    contig_names <- read.csv(bin_pathname, header=F)
    all_contigs <- NULL
    
    for (i in 1:nrow(contig_names)){
      contigname <- contig_names[i,1]
      find_contig <- classification %>%
        filter(str_detect(X..contig, contigname))
      all_contigs <- rbind(all_contigs, find_contig)
    }
    
    # majority of ORFs have to be Marinimicrobia for me to look at the bin
    all_contigs$ORFs <- str_split_fixed(all_contigs$reason, "[ ]", 4)[,3]
    all_contigs$ORFs_true <- as.numeric(str_split_fixed(all_contigs$ORFs, "[/]", 4)[,1])
    all_contigs$ORFs_all <- as.numeric(str_split_fixed(all_contigs$ORFs, "[/]", 4)[,2])
    consensus <- all_contigs %>%
      group_by(species) %>%
      summarize(support = sum(ORFs_true))
    consensus$total <- sum(all_contigs$ORFs_all)
    consensus <- consensus$species[which(consensus$support > (consensus$total / 2))]
    
    if (identical(consensus[str_detect(consensus, "Marinimicrobia")], character(0)) == F){
      a <- a + 1
      mar_bins$bin[a] <- binname
      mar_bins$sample[a] <- bin_method
    }
  }
}

mar_bins <- as.data.frame(mar_bins)
```

Use BUSCO (which is similar to CheckM to get completeness, contamination, etc of bins). One of these steps has to be run on my machine (generating the individual fasta files)

```{r}
# Get fasta files for each bin
mar_bins <- read.csv("../output/all_mar_bins.csv")
all <- readDNAStringSet("~/Library/CloudStorage/GoogleDrive-jwinter2@uw.edu/Shared drives/Rocap Lab/Project_ODZ_Marinimicrobia_Jordan/Annotations/1058_P1_2018_585_0.2um_assembly_plus.fna")
path <- "/Users/jordini/Library/CloudStorage/GoogleDrive-jwinter2@uw.edu/Shared drives/Rocap Lab/Project_ODZ_Marinimicrobia_Jordan/Bins/"

for (i in 1:nrow(mar_bins)){
  bin <- mar_bins$bin[i]
  bintype <- mar_bins$sample[i]
  bin_nms <- read.csv(paste0(path, bintype, "/", bin), header = F)
  small_fa <- str_detect(all@ranges@NAMES, paste(bin_nms$V1, collapse = "|"))
  index <- which(small_fa == T)
  small_fa <- all[index]
  writeXStringSet(small_fa, paste0("../data/Bin_fa/", bin), format = "fasta")
}
```

Run BUSCO

```{bash}
busco -i ~/github/jordan-marinimicrobia/data/Bin_fa -l bacteria_odb10 -m geno -o ~/github/jordan-marinimicrobia/output/busco_outputs -c 8
```

Combine BUSCO output table with bins summary

```{r}
busco <- read.delim("../output/batch_summary.txt")
colnames(busco)[1] <- "bin"
mar_bins <- merge(mar_bins, busco)
```

