---
title: "01-annotating-bins"
author: "Jordan Winter"
output: 
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
---

```{r setup, include=F}
library(tidyverse)
library(GEOquery)
library(Biostrings)
library(dplyr)
library(ggalluvial)
library(knitr)
opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Don't evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center" # Align plots to the center
)
```

*This document goes over the steps to check the metagenome bins for completeness and contamination.*

# Find bins that are Sulfitobacter {.unnumbered}

These bins are MAGs (metagenome assembled genomes), and I'll find which are Sulfitobacter by using the CATBAT consensus results of what each contig is. Since I know what microbe each contig is, I'll look for bins where most contigs are Roseobacteraceae, which is the family Sulfitobacter is in.

```{r}
# open file with all CATBAT results
classification <- read.csv("../data/CATBAT_all.txt", sep = "\t")

# run through each binning method
bin_path <- list.dirs("../data/Bins")
sulf_bins <- NULL
a <- 0

for (i in 2:7){
  
  pathname <- paste0(bin_path[i], "/")
  bins <- list.files(pathname)
  
  for (binname in bins){
    print(binname)
    bin_pathname <- paste0(pathname, binname)
    contig_names <- read.csv(bin_pathname, header=F)
    all_contigs <- NULL
    
    for (i in 1:nrow(contig_names)){
      contigname <- contig_names[i,1]
      find_contig <- classification %>%
        filter(str_detect(X..contig, contigname))
      all_contigs <- rbind(all_contigs, find_contig)
    }
    
    # majority of ORFs have to be Marinimicrobia for me to look at the bin
    all_contigs$ORFs <- str_split_fixed(all_contigs$reason, "[ ]", 4)[,3]
    all_contigs$ORFs_true <- as.numeric(str_split_fixed(all_contigs$ORFs, "[/]", 4)[,1])
    consensus <- all_contigs %>%
      group_by(family) %>%
      summarize(support = sum(ORFs_true)) %>%
      filter(family != "no support") # only to get all possible Mar bins
    index <- which(str_detect(consensus$family, "Roseobacteraceae"))
    consensus$family[index] <- "Roseobacteraceae"
    consensus <- consensus$family[which(consensus$support == max(consensus$support))]
    print(consensus)
    
    if (identical(consensus[str_detect(consensus, "Roseobacteraceae")], character(0)) == F){
      a <- a + 1
      sulf_bins$bin[a] <- binname
      sulf_bins$sample[a] <- basename(pathname)
    }
  }
}

sulf_bins <- as.data.frame(sulf_bins)
write.csv(sulf_bins, "../output/all_sulf_bins.csv")
```

Look at the Sulfitobacter bin output file.

```{r, eval=T}
sulf_bins <- read.csv("../output/all_sulf_bins.csv")
head(sulf_bins$bin)
```

# BUSCO {.unnumbered}

Use BUSCO (which is similar to CheckM to get completeness, contamination, etc of bins). This is run on my own computer, hence the Google Drive files.

```{r}
# Get fasta files for each bin
sulf_bins <- read.csv("../output/all_sulf_bins.csv")
all <- readDNAStringSet("~/Library/CloudStorage/GoogleDrive-jwinter2@uw.edu/Shared drives/Rocap Lab/Project_ODZ_Marinimicrobia_Jordan/Assembly/1058_P1_2018_585_0.2um_assembly_plus.fa")
path <- "../data/Bins/"

for (i in 1:nrow(sulf_bins)){
  bin <- sulf_bins$bin[i]
  bintype <- sulf_bins$sample[i]
  bin_nms <- read.csv(paste0(path, bintype, "/", bin), header = F)
  small_fa <- str_detect(all@ranges@NAMES, paste(bin_nms$V1, collapse = "|"))
  index <- which(small_fa == T)
  small_fa <- all[index]
  writeXStringSet(small_fa, paste0("../data/Bin_fa_sulf/", bin), format = "fasta")
}
```

Check output fasta files

```{r, eval=T}
fa <- readDNAStringSet("../data/Bin_fa_sulf/24_sample_bam_bin.11")
fa
```

Run BUSCO on my own computer, has to be done in a dedicated environment. I am using the generic bacteria database to search for single copy marker genes.

```{bash}
conda activate busco
busco -i github/jordan-marinimicrobia/data/Bin_fa_sulf -l bacteria_odb10 \
-m geno -o github/jordan-marinimicrobia/output/busco_outputs_sulf -c 8
```

Combine BUSCO output table with bins summary.

```{r}
busco <- read.delim("../output/busco_outputs_sulf/batch_summary.txt")
colnames(busco)[1] <- "bin"
sulf_bins <- merge(sulf_bins, busco)
write.csv(sulf_bins, "../output/all_sulf_bins.csv", row.names = F)
```

# Add sequence length to bin summary {-}

Get sequence length of all the bins and add it to bin summary table

```{r}
bin_path <- list.files("../data/Bin_fa_sulf") # annotated fasta files for each bin
lengths <- NULL

for (i in 1:length(bin_path)){
  
  bin <- bin_path[i]
  bin_fullname <- paste0("../data/Bin_fa_sulf/", bin)
  
  lengths$bin[i] <- bin
  fa <- readDNAStringSet(bin_fullname, format = "fasta")
  lengths$sum_len[i] <- sum(nchar(fa))
  
}

lengths <- as.data.frame(lengths)
sulf_bins <- merge(sulf_bins, lengths)
write.csv(sulf_bins, "../output/all_sulf_bins.csv", row.names = F)
```

Look at bins summary table.

```{r, eval=T}
sulf_bins <- read.csv("../output/all_sulf_bins.csv")
kable(head(sulf_bins, 5))
```

# Create alluvial plot {-}

Create alluvial plot to check what contigs/reads went where in each binning method using 47_sample_bam_bin.81, which is Sulfitobacter. As can be seen in the plot, most binning methods had the same contigs in them.

```{r, eval=T}
# open file
alluvial <- read.csv("../output/for_alluvial_plot.csv")

# choose starting bin
alluvial <- subset(alluvial, X47_sample_bam_bins == "47_sample_bam_bin.81")

alluvial <- pivot_longer(alluvial, cols=c(colnames(alluvial)[2:7]), names_to="bin_method", values_to="bin")
alluvial <- subset(alluvial, is.na(bin) == F)
alluvial$bin_method <- factor(alluvial$bin_method, levels = c("X24_sample_bam_bins","X47_sample_bam_bins", "short_reads_bam_bins",
  "transcriptomes_bam_bins","assembly_plus_bins","assembly_bins"))

ggplot(alluvial,
       aes(x = bin_method, stratum = bin, alluvium = contig_name,
           fill = bin, label = bin)) +
  geom_flow(stat = "alluvium", lode.guidance = "frontback",
            color = "darkgray") +
  geom_stratum() +
  theme_bw() +
  theme(axis.text.x = element_text(angle = 45, vjust=0.5)) +
  ggtitle("Sulfitobacter Contigs Using Varied Binning Methods")
```
