*This will be the working document for my project.*

Load required packages.

```{r}
library(tidyverse)
library(GEOquery)
```

Import data. Note that this data is not publicly available yet, so I'm getting it from my Google Drive.

```{r}
# set path to all data
path <- "~/Library/CloudStorage/GoogleDrive-jwinter2@uw.edu/Shared drives/Rocap Lab/Project_ODZ_Marinimicrobia_Jordan/"
```

Find bins that are Marinimicrobia. These bins are MAGs (metagenome assembled genomes), and I'll find which are Marinimicrobia by using the CATBAT consensus results of what each contig is. Since I know what microbe each contig is, I'll look for bins where the majority of contigs are Marinimicrobia.

```{r}
# open file
classification <- read.csv(paste0(path, "CAT-BAT/1058_P1_2018_585_0.2um/1058_P1_2018_585_0.2um_assembly_plus.contig2classification_names.txt"), sep = "\t")

# get only those labeled as Marinimicrobia
classification <- classification[str_detect(classification$species, "Marinimicrobia"),]

# add this classification file to the repo
write.csv(classification, "~/github/jordan-marinimicrobia/output/CATBAT_marinimicrobia.csv")
```

Now that I have a classification file and my bins, I can find bins where the majority of reads and contigs are Marinimicrobia

```{r}
# get Marinimicrobia bins
classification <- read.csv("~/github/jordan-marinimicrobia/data/CATBAT_all.txt", sep = "\t")

# run through each binning method
bin_path <- "~/github/jordan-marinimicrobia/data/Bins/"
  
mar_bins <- NULL
a <- 0

for (i in 1:6){
  
  bin_method <- list.files(bin_path)[i]
  pathname <- paste0(bin_path, bin_method, "/")
  bins <- list.files(pathname)
  
  for (binname in bins){
    print(binname)
    bin_pathname <- paste0(pathname, binname)
    contig_names <- read.csv(bin_pathname, header=F)
    all_contigs <- NULL
    
    for (i in 1:nrow(contig_names)){
      contigname <- contig_names[i,1]
      find_contig <- classification %>% filter(str_detect(X..contig, contigname))
      all_contigs <- rbind(all_contigs, find_contig)
    }
    
    # majority has to be Marinimicrobia for me to look at the bin
    consensus <- names(which.max(table(all_contigs$species)))
    if (identical(consensus[str_detect(consensus, "Marinimicrobia")], character(0)) == F){
      a <- a + 1
      mar_bins$bin[a] <- binname
      mar_bins$sample[a] <- bin_method
    }
  }
  
}

mar_bins <- as.data.frame(mar_bins)
```

Add contig and read length together to get bin size

```{r}
lengths <- read.csv("~/github/jordan-marinimicrobia/data/contig_lengths.txt", sep = "\t", header = F)
colnames(lengths) <- c("sample", "contig_name", "length")
mar_bins <- read.csv("~/github/jordan-marinimicrobia/output/all_mar_bins.csv")
mar_bins$total_length <- NULL
path <- "~/github/jordan-marinimicrobia/data/Bins/"

for (i in 1:nrow(mar_bins)){
  bin_number <- mar_bins$bin[i]
  folder <- mar_bins$sample[i]
  pathname <- paste0(path, folder, "/", bin_number)
  contigs <- read.csv(pathname, header=F)
  colnames(contigs) <- "contig_name"
  bin_length <- merge(contigs, lengths)
  total <- sum(bin_length$length)
  mar_bins$total_length[i] <- total
}
```

