---
title: "03-ko"
author: "Jordan Winter"
output: 
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
---

```{r setup, include=F}
library(tidyverse)
library(knitr)
library(pheatmap)
opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Don't evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center" # Align plots to the center
)
```

*This document details Kegg pathways and creation of the presence/absence plot*

# Get KO annotations {-}

Get Kegg orthologs for variable regions in the pangenome. This is done in anvio by using the "summarize" option in anvi interactive.

# Get specific Kegg pathways {-}

Annotate contigs with general metabolisms of interest.

```{r}
genes <- read.csv("../data/KO_numbers_path.csv")
colnames(genes) <- c("accession", "pathway")
ko <- merge(ko, genes)

# count how many genes there are of each pathway
all_paths <- unique(genes$pathway)
pathway <- NULL

for (one_contig in unique(ko$contig)){
  ko_small <- subset(ko, contig == one_contig)
  results <- NULL
  for (i in 1:length(all_paths)){
    path <- all_paths[i]
    results$count[i] <- length(which(ko_small$pathway == path))
    results$path[i] <- path
  }
  results <- as.data.frame(results)
  results$bin <- one_contig
  pathway <- rbind(pathway, results)
}

pathway$presence <- "no"
index <- which(pathway$count > 0)
pathway$presence[index] <- "yes"
```

# Presence-absence plot of pathways {-}

Plot genes of interest

```{r}
# presence-absence
a <- ggplot(pathway, aes(bin, path, fill=presence)) +
  geom_tile(color="black") +
  scale_x_discrete(guide = guide_axis(n.dodge = 2)) +
  scale_fill_grey(start = 1, end = 0.5) +
  theme_bw()
a
```

# Plot of specific aerobic genes {-}

```{r, eval=T}
genes <- read.csv("~/Downloads/sulf_genomes/sulfitobacter/SUMMARY_variable_region/sulfitobacter_gene_clusters_summary.txt", sep = "\t")

ko_of_interest <- read.csv("~/Downloads/sulf_genomes/KO_of_interest.csv")

# choose genes you want to plot
ko_of_interest <- subset(ko_of_interest, pathway == "terminal oxidase")

index <- str_detect(genes$KOfam_ACC, paste(ko_of_interest$ko, collapse = "|"))
genes_small <- genes[which(index),]

a <- ggplot(genes_small) +
  geom_bar(aes(KEGG_Module, fill = bin_name)) +
  theme_bw() +
  facet_wrap(~genome_name) +
  scale_x_discrete(guide = guide_axis(angle = 90))
a
```

# Presence-absence plot for everything

```{r, eval=T}
summary <- read.csv("~/Downloads/sulf_genomes/combined/pangenome/SUMMARY_variable/pangenome_gene_clusters_summary.txt", sep = "\t")
summary <- summary[,c("genome_name", "KOfam_ACC")]
summary <- subset(summary, KOfam_ACC != "")

test <- as.matrix(table(summary$genome_name, summary$KOfam_ACC))

a <- pheatmap(test, show_colnames=F)
a
```
