---
title: "03-ko"
author: "Jordan Winter"
output: 
  html_document:
    theme: cosmo
    toc: true
    toc_float: true
    number_sections: true
    code_folding: show
---

```{r setup, include=F}
library(tidyverse)
library(knitr)
library(pheatmap)
library(DT)
opts_chunk$set(
  echo = TRUE,         # Display code chunks
  eval = FALSE,        # Don't evaluate code chunks
  warning = FALSE,     # Hide warnings
  message = FALSE,     # Hide messages
  fig.width = 6,       # Set plot width in inches
  fig.height = 4,      # Set plot height in inches
  fig.align = "center" # Align plots to the center
)
```

*This document details Kegg pathways and creation of the presence/absence plot*

# Get summary stats on genomes {.unnumbered}

Get Kegg orthologs for all regions in the pangenome. This is done in anvio by using the "summarize" option in anvi interactive.

```{r, eval=T}
summary <- read.csv("../output/pangenome_gene_clusters_summary.txt", sep = "\t")
summary <- summary[,3:18]

datatable(summary)
```

# Plot of specific aerobic genes {.unnumbered}

```{r, eval=T}
genes <- read.csv("~/Downloads/sulf_genomes/sulfitobacter/SUMMARY_variable_region/sulfitobacter_gene_clusters_summary.txt", sep = "\t")

ko_of_interest <- read.csv("~/Downloads/sulf_genomes/KO_of_interest.csv")

# choose genes you want to plot
ko_of_interest <- subset(ko_of_interest, pathway == "terminal oxidase")

index <- str_detect(genes$KOfam_ACC, paste(ko_of_interest$ko, collapse = "|"))
genes_small <- genes[which(index),]

genes <- ggplot(genes_small) +
  geom_bar(aes(KEGG_Module, fill = bin_name)) +
  theme_bw() +
  facet_wrap(~genome_name) +
  scale_x_discrete(guide = guide_axis(angle = 90))
genes
```

# Heatmap for different genes {.unnumbered}

```{r, eval=T}
summary <- read.csv("../output/pangenome_gene_clusters_summary.txt", sep = "\t")
summary <- summary[,c("genome_name", "KOfam_ACC")]
summary <- subset(summary, KOfam_ACC != "" & genome_name != "sulf_ctg_121" & genome_name != "sulf_ctg_16")

heat_matrix <- as.matrix(table(summary$genome_name, summary$KOfam_ACC))

#heatmap_all <- pheatmap(heat_matrix, show_colnames=F)

index <- NULL
for (i in 1:ncol(heat_matrix)){
  if (length(unique(heat_matrix[,i])) > 1){
    index[i] <- TRUE
  } else {
    index[i] <- FALSE
  }
}
matrix_small <- heat_matrix[,which(index == TRUE)]

heatmap <- pheatmap(matrix_small, show_colnames=F)
heatmap
```

# Compute genome similarity {.unnumbered}

```{r, engine='bash'}
anvi-compute-genome-similarity --external-genomes sulf_external_genomes.txt --program pyANI --output-dir combined/ANI --num-threads 4 --pan-db combined/pangenome/pangenome-PAN.db
```

![average nucleotide identity at the 95% level](../output/sulf-ani.png)

# Genome synteny {-}

```{r, engine='bash'}
anvi-analyze-synteny -g sulf-all-GENOMES.db --annotation-source KOfam --ngram-window-range 2:3 -o ngrams -p pangenome/pangenome-PAN.db --ngram-source gene_clusters
```

```{r}
ngrams <- read.csv("~/Downloads/sulf_genomes/combined/ngrams", sep = "\t")
```
